src/SprykerEco/Zed/Optivo/Business/OptivoBusinessFactory.php

```diff
@@ -1,86 +1,211 @@
-     * @return \SprykerEco\Zed\Optivo\Business\Api\OptivoApiInterface
+     * @return \SprykerEco\Zed\Optivo\Business\Handler\Order\OrderEventHandlerInterface
      */
-    public function createOptivoApi()
+    public function createNewOrderEventHandler(): OrderEventHandlerInterface
     {
-        return new OptivoApi(
+        return new OrderEventHandler(
+            $this->createNewOrderMapper(),
             $this->createOptivoApiAdapter(),
-            $this->getConfig()
+            $this->getSalesFacade()
         );
     }
 
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Handler\Customer\CustomerEventHandlerInterface
+     */
+    public function createCustomerEventHandler(): CustomerEventHandlerInterface
+    {
+        return new CustomerEventHandler($this->createCustomerMapper(), $this->createOptivoApiAdapter());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Handler\Customer\CustomerEventHandlerInterface
+     */
+    public function createNewsletterSubscriptionEventHandler(): CustomerEventHandlerInterface
+    {
+        return new CustomerEventHandler($this->createCustomerNewsletterMapper(), $this->createOptivoApiAdapter());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToMoneyFacadeInterface
+     */
+    protected function getMoneyFacade(): OptivoToMoneyFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_MONEY);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToLocaleFacadeInterface
+     */
+    protected function getLocaleFacade(): OptivoToLocaleFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_LOCALE);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Dependency\Facade\OptivoToSalesFacadeInterface
+     */
+    protected function getSalesFacade(): OptivoToSalesFacadeInterface
+    {
+        return $this->getProvidedDependency(OptivoDependencyProvider::FACADE_SALES);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Request\RequestUrlBuilderInterface
+     */
+    protected function createRequestUrlBuilder(): RequestUrlBuilderInterface
+    {
+        return new RequestUrlBuilder($this->getConfig());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Response\ResponseConverterInterface
+     */
+    protected function createResponseConverter(): ResponseConverterInterface
+    {
+        return new ResponseConverter();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Optivo\Business\Api\Adapter\Http\HttpAdapterInterface
+     */
+    protected function createGuzzleAdapter(): HttpAdapterInterface
+    {
+        return new Guzzle($this->getConfig());
+    }
+
     /**
      * @return \SprykerEco\Zed\Optivo\Business\Api\Adapter\OptivoApiAdapterInterface
      */
-    protected function createOptivoApiAdapter()
+    protected function createOptivoApiAdapter(): OptivoApiAdapterInterface
```

Comment:
All methods in factories have to be public.

Reply:
done